Code.gs

```diff
@@ -279,19 +312,26 @@ function ParseNotificationTime(notificationString){
 function sameEvent(x){
   return x.id == this;
 }
-//function EventExists(calendar, event){
-//
-//  var events = calendar.getEvents(event.startTime, event.endTime, {search : event.id});
-//  return events.length > 0;
-//}
-
-function GetUTCTime(parameter){
-  parameter = parameter.substr(1); //Remove leading ; or : character
-  if (parameter.includes("TZID")){
-    var tzid = parameter.split("TZID=")[1].split(":")[0];
-    var time = parameter.split(":")[1];
-    return Moment.moment.tz(time,tzid).tz("Etc/UTC").format("YYYYMMDDTHHmmss") + "Z";
+
+function CheckForUpdate(){
+  var alreadyAlerted = PropertiesService.getScriptProperties().getProperty("alertedForNewVersion");
+  if (alreadyAlerted == null){
+    try{
+      var thisVersion = 1.0;
+      var html = UrlFetchApp.fetch("https://github.com/derekantrican/GAS-ICS-Sync/releases");
+      var regex = RegExp("<a.*title=\"\\d\\.\\d\">","g");
+      var latestRelease = regex.exec(html)[0];
+      regex = RegExp("\"(\\d.\\d)\"", "g");
+      var latestVersion = Number(regex.exec(latestRelease)[1]);
+      
+      if (latestVersion > thisVersion){
+        if (email != ""){
+          GmailApp.sendEmail(email, "New version of GAS-ICS-Sync is available!", "There is a new version of \"GAS-ICS-Sync\". You can see the latest release here: https://github.com/derekantrican/GAS-ICS-Sync/releases");
+        
+          PropertiesService.getScriptProperties().setProperty("alertedForNewVersion", true);
```

Comment:
I see a problem here, as this script property is never reset (e.g. during `install()`) so I expect the update check to work _only once_ and then always fail because `alertedForNewVersion` stays true.

As a solution I would suggest to store the last _version number_ we alerted for, instead of only a boolean value. This way the user will be alerted again if a newer new version is available 😄 

Suggested new implementation:
```js
function checkForUpdate(){
  // No need to check if we can't alert anyway
  if (email == "")
    return;

  var lastAlertedVersion = PropertiesService.getScriptProperties().getProperty("alertedForNewVersion");
  try {
    var thisVersion = 5.4;
    var latestVersion = getLatestVersion();

    if (latestVersion > thisVersion && latestVersion != lastAlertedVersion){
      GmailApp.sendEmail(email,
        "New version of GAS-ICS-Sync is available!",
        "There is a new version of \"GAS-ICS-Sync\". You can see the latest release here: https://github.com/derekantrican/GAS-ICS-Sync/releases");

      PropertiesService.getScriptProperties().setProperty("alertedForNewVersion", latestVersion);
    }
  }
  catch (e){}

  function getLatestVersion(){
    var html = UrlFetchApp.fetch("https://github.com/derekantrican/GAS-ICS-Sync/releases");
    var regex = RegExp("<a.*title=\"\\d\\.\\d\">", "g");
    var latestRelease = regex.exec(html)[0];
    regex = RegExp("\"(\\d.\\d)\"", "g");
    return Number(regex.exec(latestRelease)[1]);
  }
}
```

Also, the comment for the `email` variable in Code.gs needs to be updated from:  
`// OPTIONAL: If "emailWhenAdded" is set to true, you will need to provide your email`
to  
`// OPTIONAL: If "emailSummary" is set to true or you want to receive notifications when a new version is available, you will need to provide your email address`

Reply:
Yeah, this was added that way because users can run this script quite often (maybe every 5 minutes) and we don't want to start sending someone an email every 5 minutes that a new version is available.

If I understand correctly, what you're suggestion is the current workflow (eg user has version 5 and we alert them once that version 6 has come out) but with the addition that we alert for every new version (eg user has version 5, we alerted them for version 6, they still haven't upgraded and version 7 comes out and we alert them again)?

Reply:
Exactly, that's what my change is intended to achieve.

But I also just understood that a user will usually "upgrade" to the new version by copying the script from your account to theirs again (so they have a new project and `alertedForNewVersion` will be `false` for that new project.

So with the current code every user gets one update notification unless they actually update, and that's totally fine for me (I just thought that _even if_ I updated, the script property would stay true thus preventing all but the first ever notification. That was a misunderstanding of the update process on my side.)

The `email` comment "issue" still stands, though 😁

Reply:
Yes, exactly - the variable is reset when the new script is copied (provided they use the "Copy to my Google Drive" option and don't just copy and replace the code)

What is "the `email` comment "issue""? I think I missed that

Reply:
I just went ahead and created #163 because a diff says more than a thousand words 😄